tree/tree/src/TBasket.cxx

```diff
@@ -932,6 +932,9 @@ Int_t TBasket::WriteBuffer()
       return -1;
    }
    fMotherDir = file; // fBranch->GetDirectory();
+#ifdef R__USE_IMT
```

Comment:
Given the slightly unusual use pattern (take the lock then explicit unlock then lock).  Can you add comment on what this is propecting (and why it will not provoke a dead lock).